### ISSUE TITLE
prototype norm problems 

### ISSUE THREAD

**Issue by zkmake520** [facebookresearch/swav#16]
Hi  @mathildecaron31 
As you said,
In the paper, the prototypes are indeed normalized along the first dimension because the prototypes matrix, `C`, is of dimension DxK (i.e, 128x3000).

On the contrary, in the code, `w` is of dimension KxD (i.e, 3000x128). You can easily check that the normalization is done correctly by printing:
```
print(torch.norm(w, dim=1).shape) # should give 3000
print(torch.norm(w, dim=1)) # should give a vector with 1 everywhere
``` 

But the the code goes:
    ```
     self.prototypes = nn.Linear(output_dim, nmb_prototypes, bias=False)
    ```
    which output_dim = 128 , nmb_prototypes = 3000
 So the `w` is of dimension D * K (128 * 3000),  which means there are 3000 prototypes and each prototype is a vector with dim 128

Thanks

**Comment by mathildecaron31** [comment]
You can print the dimension of `w` with `print(w.shape)` to check that it is of dimension K*D. Each row of `w` is a prototype of dimension 128.

**Comment by mathildecaron31** [comment]
See that Linear.weight is of shape (out_features,in_features) 
https://pytorch.org/docs/stable/generated/torch.nn.Linear.html#torch.nn.Linear
